databroker/core.py

```diff
@@ -64,31 +64,41 @@ class Document(dict):
     consumers that expect an object that satisfies isinstance(obj, dict).
     This implementation detail may change in the future.
     """
-    __slots__ = ()
-    _NOT_MUTABLE_MSG = (
-        "Documents are not mutable. Call the method to_dict() to make a "
-        "fully independent and mutable copy.")
+    __slots__ = ('__not_a_real_dict',)
 
-    def __setitem__(self, k, v):
-        raise NotMutable(self._NOT_MUTABLE_MSG)
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # This lets pickle recognize that this is not a literal dict and that
+        # it should respect its custom __setstate__.
+        self.__not_a_real_dict = True
 
-    def __delitem__(self, k):
-        raise NotMutable(self._NOT_MUTABLE_MSG)
-
-    def pop(self, k):
-        raise NotMutable(self._NOT_MUTABLE_MSG)
-
-    def popitem(self):
-        raise NotMutable(self._NOT_MUTABLE_MSG)
+    def __getstate__(self):
+        return dict(self), self.__not_a_real_dict
```

Comment:
If `self.__not_a_real_dict` will always be `True`, I don't think we need to include it in the serialization.